ros_buildfarm/config/ci_build_file.py

```diff
@@ -123,3 +123,14 @@ def __init__(self, name, data):  # noqa: D107
                     assert isinstance(plot_config_data, dict)
                     self.show_plots[plot_group].append(
                         PlotConfig(name, plot_config_data))
+
+        self.benchmark_patterns = []
+        if 'benchmark_patterns' in data:
+            self.benchmark_patterns = data['benchmark_patterns']
+            assert isinstance(self.benchmark_patterns, list)
+            assert all(isinstance(pattern, str) for pattern in self.benchmark_patterns)
+
+        self.benchmark_schema = None
+        if 'benchmark_schema' in data:
+            self.benchmark_schema = data['benchmark_schema']
+            assert self.benchmark_patterns
```

Comment:
Adding a message would be helpful in case the assertion triggers.

Reply:
9f24017